### ISSUE TITLE
Info `:file` returns temporary file in boot for cljs

### ISSUE THREAD

**Issue by SevereOverfl0w** [clojure-emacs/cider-nrepl#421]
## Expected behavior
`{:op :info :symbol "some.cljs/symbol" :session "cljs-session"}` contains a "file" like: `~/project/src/some/cljs.cljs`
## Actual behavior
`{:op :info :symbol "some.cljs/symbol" :session "cljs-session"}` contains a "file" like: `~/.boot/cache/project/-adfjdkj/k1j/src/some/cljs.cljs`
## Steps to reproduce the problem

1. `$ boot -d boot/new new -t tenzing -n your-app`
2. `$ boot cider dev` (or however you run `dev` with cider!)
3. Connect to clojurescript repl & open http://localhost:3000
4. Open src/cljs/your_app/app.cljs
5. Jump to source on `init` (I did it on the definition, this is potentially editor specific!)
6. You're in a temporary file!

### cider-nrepl version

0.15.0-SNAPSHOT

### Java version

1.8

### Operating system

Linux

**Comment by bbatsov** [comment]
That's a side-effect of how boot works. We have discussed this a few times, but I don't remember what was decided. Obviously translating the filenames is not hard and can be done.

**Comment by SevereOverfl0w** [comment]
This seems to be working for me locally:

```
diff --git a/src/cider/nrepl/middleware/info.clj b/src/cider/nrepl/middleware/info.clj
index 57efbb2..5ce9d6e 100644
--- a/src/cider/nrepl/middleware/info.clj
+++ b/src/cider/nrepl/middleware/info.clj
@@ -79,7 +79,23 @@
 (defn info-cljs
   [env symbol ns]
   (some-> (cljs-info/info env symbol ns)
-          (select-keys [:file :line :ns :doc :column :name :arglists])))
+          (select-keys [:file :line :ns :doc :column :name :arglists])
+          (update
+            :file
+            (fn [f]
+              (let [path (java.nio.file.Paths/get f (into-array String []))
+                    path-count (.getNameCount path)]
+                (first
+                  (sequence
+                    (comp (map #(.subpath path % path-count))
+                          (map str)
+                          (filter
+                            (fn [subpath]
+                              (let [r (clojure.java.io/resource subpath)]
+                                (and r
+                                     (= (.getAbsolutePath (clojure.java.io/file r))
+                                        f))))))
+                    (range path-count))))))))
 
 (defn info-java
   [class member]
```

Would a patch to this approach be accepted? (with some cleanup)

**Comment by bbatsov** [comment]
Fixed by #422.
